packages/legend-application-query/src/stores/QueryEditorStore.ts

```diff
@@ -527,28 +247,19 @@ export abstract class QueryEditorStore {
 
   readonly initState = ActionState.create();
 
-  queryBuilderState?: QueryBuilderState | undefined;
-  saveAsState?: QuerySaveAsState | undefined;
-  saveState?: QuerySaveState | undefined;
-  renameState?: QueryRenameState | undefined;
-  title: string | undefined;
+  _queryBuilderState?: QueryBuilderState | undefined;
+  queryCreatorState: QueryCreatorState;
   existingQueryName: string | undefined;
 
   constructor(
     applicationStore: LegendQueryApplicationStore,
     depotServerClient: DepotServerClient,
   ) {
     makeObservable(this, {
```

Comment:
should we add `_queryBuilderState: observable` and `queryBuilderState: computed`?